src/app/shared/models/demand.ts

```diff
@@ -0,0 +1,34 @@
+import { Beneficiary } from './beneficiary';
+import { IVolunteer } from './volunteers';
+
+export enum DemandType {
+  warm_lunch = 'Warm Lunch',
+  grocery = 'Grocery',
+  medicine = 'Medicine',
+  invoices = 'Pay invoices',
+  transport = 'Transport person',
```

Comment:
Can we have here the value to be the same as key? So we can translate them.

Reply:
It totally ok for translation - you can use in a component mappings like
demands=Object.entries(DemandType).map(([key, value]) => key);
Such way you obtain array of keys (as you requested) and have back-end compliance definition in front-end constants.ts.

Reply:
the funny think - you have default language naming inside enum simply by using
demands=Object.entries(DemandType).map(([key, value]) => value); 
:)

Reply:
Is also needed for mapping to the enum coming from the backend.

Reply:
sorry - probably I misunderstand something, but this example looks not clear for me. 
for translation reason (like i18 for example) you simply use necessary value tooked from translationfile.ro/ru/en by enum key ...

Reply:
Yes, you can, but is not a straight forward solution. another example

```typescript jsx
const aListOfDemantTypes = [
  DemandType.warm_lunch,
  DemandType.grocery,
  DemandType.medicine,
]
```

And you want to show them in frontend. You can't localize them, because they are

```typescript jsx
const aListOfDemantTypes = [
  'Warm Lunch',
  'Grocery',
  'Medicine',
]
```

Reply:
In source code there should not be any strings that needs to be showed in UI. Everything should be in translation files.

Reply:
> 
> 
> In source code there should not be any strings that needs to be showed in UI. Everything should be in translation files.

Yes! totally agree with you - that's why you use keys as references in the code and loaded lang values for view, if lang file somehow wasn't loaded you simply use value from initial enum to show.

Reply:
Here a question. This comes from backend, we want to show user roles in UI, how do we translate them?

```typescript jsx
const user: User = {
  roles: ['administrator', 'operator']
}
```

Reply:
Bad example again )

Reply:
finally agree with the "bug" in back-end :)

Reply:
So now we need to make enum identical to that